salesgpt/agents.py

```diff
@@ -105,21 +106,22 @@ def step(self, stream: bool = False):
             streaming generator object to manipulate streaming chunks in downstream applications.
         """
         if not stream:
-            self._call(inputs={})
+            return self._call(inputs={})#delete return?
         else:
             return self._streaming_generator()
 
     @time_logger
-    def astep(self, stream: bool = False):
+    async def astep(self, stream: bool = False):
         """
         Args:
             stream (bool): whether or not return
             streaming generator object to manipulate streaming chunks in downstream applications.
         """
         if not stream:
-            self._acall(inputs={})
+            await self._acall(inputs={})
```

Comment:
not sure if this is awaitable, make sure it's correct, it may also be why some tests are failing.